SpatialGDK/Source/SpatialGDKEditor/Private/Utils/TransientUObjectEditor.cpp

```diff
@@ -0,0 +1,167 @@
+void UTransientUObjectEditor::LaunchTransientUObjectEditor(const FString& EditorName, UClass* ObjectClass)
+{
+	if (!ObjectClass)
+	{
+		return;
+	}
+
+	if (!ObjectClass->IsChildOf<UTransientUObjectEditor>())
+	{
+		//UE_LOG(LogSpatialGDKTransientUObjectEditor, Error, TEXT("Class %s is not a class compatible with TransientUObjectEditor"), *ObjectClass->GetName());
+		return;
+	}
+
+	UTransientUObjectEditor* ObjectInstance = NewObject<UTransientUObjectEditor>(GetTransientPackage(), ObjectClass);
+	ObjectInstance->AddToRoot();
+
+	FPropertyEditorModule& PropertyEditorModule = FModuleManager::LoadModuleChecked<FPropertyEditorModule>("PropertyEditor");
+
+	TArray<UObject*> ObjectsToView;
+	ObjectsToView.Add(ObjectInstance);
+
+	FDetailsViewArgs Args;
+	Args.bHideSelectionTip = true;
+	Args.bLockable = false;
+	Args.bAllowSearch = false;
+	Args.bShowPropertyMatrixButton = false;
+
+	TSharedRef<IDetailsView> DetailView = PropertyEditorModule.CreateDetailView(Args);
+
+	bool bHaveTemplate = false;
+	for (int32 i = 0; i < ObjectsToView.Num(); i++)
+	{
+		if (ObjectsToView[i] != NULL && ObjectsToView[i]->IsTemplate())
+		{
+			bHaveTemplate = true;
+			break;
+		}
+	}
+
+	DetailView->SetIsPropertyVisibleDelegate(FIsPropertyVisible::CreateStatic(&ShouldShowProperty, bHaveTemplate));
+
+	DetailView->SetObjects(ObjectsToView);
+
+	TSharedRef<SVerticalBox> VBoxBuilder = SNew(SVerticalBox)
+		+ SVerticalBox::Slot()
+		.AutoHeight()
+		.FillHeight(1.0)
+		[
+			DetailView
+		];
+
+
+	// Expand propeprty view.
+	// Implementation not accessible :(
+	//if (SDetailsView* DetailViewImpl = Cast<SDetailsView>(DetailView))
+	//{
+	//	DetailViewImpl->SetRootExpansionStates(true, false);
+	//}
+
+	// Add UFunction marked Exec as buttons in the editor's window
+	for (TFieldIterator<UFunction> FuncIt(ObjectClass); FuncIt; ++FuncIt)
+	{
+		UFunction* Function = *FuncIt;
+		if (Function->HasAnyFunctionFlags(FUNC_Exec) && (Function->NumParms == 0))
+		{
+			const FString FunctionName = Function->GetName();
+			const FText ButtonCaption = FText::FromString(FunctionName);
+			const FString FilterString = FunctionName;
+
+			VBoxBuilder->AddSlot()
+				.AutoHeight()
+				.VAlign(VAlign_Bottom)
+				.HAlign(HAlign_Right)
+				.Padding(2.0)
+				[
+					SNew(SHorizontalBox)
+					+ SHorizontalBox::Slot()
+				.AutoWidth()
+				.Padding(2.0)
+				[
+					SNew(SButton)
+					.Text(ButtonCaption)
+				.OnClicked(FOnClicked::CreateStatic(&ExecuteEditorCommand, ObjectInstance, Function))
+				]
+				];
+		}
+	}
+
+	TSharedRef<SWindow> NewSlateWindow = SNew(SWindow)
+		.Title(FText::FromString(EditorName))
+		//.AutoCenter(EAutoCenter::PrimaryWorkArea)
+	  //.SizingRule(ESizingRule::Autosized)
+		[
+			SNew(SBorder)
+			.BorderImage(FEditorStyle::GetBrush(TEXT("PropertyWindow.WindowBorder")))
+			[
+				VBoxBuilder
+			]
+		];
+
+	// If the main frame exists parent the window to it
```

Comment:
What is the main frame here? The editor window or something?

Reply:
I guess, that's code I extracted from the property editor module. I just assume it's doing the right thing.